modules/fatfs/zfs_ffsystem.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * OS Dependent Functions for FatFs
+ *
+ * Copyright (c) 2023 Husqvarna AB
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/* The file is based on template file by (C)ChaN, 2018, as
+ * available from FAT FS module source:
+ * https://github.com/zephyrproject-rtos/fatfs/blob/master/option/ffsystem.c
+ */
+
+#include <ff.h>
+
+#if FF_USE_LFN == 3	/* Dynamic memory allocation */
+/* Allocate a memory block */
+void *ff_memalloc(UINT msize)
+{
+	return k_malloc(msize);
+}
+
+/* Free a memory block */
+void ff_memfree(void *mblock)
+{
+	k_free(mblock);
+}
+#endif /* FF_USE_LFN == 3 */
+
+#if FF_FS_REENTRANT	/* Mutual exclusion */
+static struct k_mutex Mutex[FF_VOLUMES + 1]; /* Table of Zephyr mutex */
```

Comment:
Where the + 1 comes from?

Reply:
The + 1 is for the system mutex. Same as in the template file by ChaN. I added a comment about it.